src/pixelator/pna/pixeldataset/io/__init__.py

```diff
@@ -107,13 +107,44 @@ def __exit__(self, exc_type, exc_value, traceback):
         """Close the writer context manager."""
         self.close()
 
-    def _write_parquet_file_to_table(self, table_name, edgelist_file: Path):
-        self._connection.sql(
-            f"""
-            CREATE OR REPLACE TABLE {table_name} AS SELECT * FROM read_parquet($parquet_file);
-            """,
-            params={"parquet_file": str(edgelist_file)},
-        )
+    def _write_parquet_file_to_table(
```

Comment:
I think we could probably get of a performance improvement and more readable code here if we just pass a list of parquet files instead of doing the insert operations. See the docs here: https://duckdb.org/docs/stable/data/multiple_files/overview.html#parquet

That would give us the func signature here: `def _write_parquet_file_to_table(self, table_name: str, edgelist_file: Path | list[Path]): ...`